### PR TITLE
SW-2695 Generic Pre-Exposed Filter

### PR DESCRIPTION
**Pull request by charlie-tf**
- created the filter component using MultiSelect
- implemented the filter in Inventory
- update style for mobile
- [WIP] wrote some tests, but having trouble getting them to run :(
  - (I'll get these working and have a followup PR)

image
image
image
image

### CODE REVIEW

File: src/components/Inventory/InventoryFiltersPopover.tsx [hunk 1]
```diff
@@ -86,96 +69,56 @@ export default function InventoryFiltersPopover({ filters, setFilters }: Invento
...
-      </Popover>
+      <div className={classes.dropdown} onClick={handleClick}>
+        <Typography>{strings.NURSERIES}</Typography>
+        <Icon name={Boolean(anchorEl) ? 'chevronUp' : 'chevronDown'} className={classes.dropdownIconRight} />
+      </div>
+      {isMobile && Boolean(anchorEl) ? (
+        <div className={classes.mobileContainer}>{renderFilterMultiSelect()}</div>
+      ) : (
+        <Popover
+          id='pre-exposed-filter-popover'
```
Comment (karthikbtf) [inline src/components/Inventory/InventoryFiltersPopover.tsx:105]: add `-inventory-` somewhere in there?

File: src/components/Inventory/Search.tsx [hunk 2]
```diff
@@ -46,7 +52,7 @@ export default function Search(props: SearchProps): JSX.Element {
         <InventoryFilters filters={filters} setFilters={setFilters} />
       </Box>
       <Grid xs={12} display='flex'>
-        <PillList data={filterPillData} onRemove={(id: number) => removeFilter(id, setFilters)} />
+        <PillList data={filterPillData ?? []} onRemove={(_) => setFilters({})} />
```
Comment (karthikbtf) [inline src/components/Inventory/Search.tsx:55]: could skip the '_' maybe?

File: src/components/common/FilterMultiSelect/index.tsx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    width: (props: StyleProps) => (props.isMobile ? '100%' : 'auto'),
+  },
+  multiSelectStyle: {
+    height: '100%',
+    width: '100%',
+  },
+}));
+
+type FilterMultiSelectProps<T> = {
+  filterLabel: string;
```
Comment (karthikbtf) [inline src/components/common/FilterMultiSelect/index.tsx:23]: label should suffice?

File: src/components/common/FilterMultiSelect/index.tsx [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+  initialSelection: T[];
+  onCancel: (finalSelection: T[]) => void;
```
Comment (karthikbtf) [inline src/components/common/FilterMultiSelect/index.tsx:25]: do we need the finalSelection in onCancel?

File: src/components/common/FilterMultiSelect/index.tsx [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+  };
+
+  const onRemove = (item: T) => {
+    const index = selection.findIndex((x) => item === x);
+    const newSelection = [...selection];
+    newSelection.splice(index, 1);
+    setSelection(newSelection);
+  };
+
+  const clearFilters = () => {
```
Comment (karthikbtf) [inline src/components/common/FilterMultiSelect/index.tsx:56]: interesting.. i like this behavior more.
in the current UX, reset applies the cleared-filters in the parent.. it's not a transient function. i think that's a bug.. cc @constanzauanini ?

Comment (karthikbtf) [reply src/components/common/FilterMultiSelect/index.tsx:56]: btw, if you hit reset and cancel and reopen the filter box,, do we see the original state?

**Review by karthikbtf — APPROVED** [review]
this is super cool!! has been very methodical so far, right from the web-components changes.. easy to follow.
a few comments

File: src/components/common/FilterMultiSelect/index.tsx (hunk 5 above)
Comment (karthikbtf) [reply src/components/common/FilterMultiSelect/index.tsx:56]: See reset behavior in withdrawal log filters

File: src/components/Inventory/Search.tsx (hunk 2 above)
Comment (charlie-tf) [reply src/components/Inventory/Search.tsx:55]: oh yeah! i forgot that's allowed

File: src/components/common/FilterMultiSelect/index.tsx (hunk 5 above)
Comment (karthikbtf) [reply src/components/common/FilterMultiSelect/index.tsx:56]: the reset/cancel works in vercel

File: src/components/common/FilterMultiSelect/index.tsx (hunk 4 above)
Comment (charlie-tf) [reply src/components/common/FilterMultiSelect/index.tsx:25]: not necessarily. I'm ignoring it in the Inventory implementation. I guess I can't think of a case where one would want it, so I'll take it out.

File: src/components/common/FilterMultiSelect/index.tsx (hunk 5 above)
Comment (charlie-tf) [reply src/components/common/FilterMultiSelect/index.tsx:56]: >if you hit reset and cancel and reopen the filter box,, do we see the original state?

yup!

File: src/components/common/FilterMultiSelect/index.tsx (hunk 4 above)
Comment (karthikbtf) [reply src/components/common/FilterMultiSelect/index.tsx:25]: that might be good, typically cancel is a void behavior.. parent doesn't need to know of any changes.

**Review by karthikbtf — APPROVED** [review]
Thanks for the changes 🙌

File: src/components/common/FilterMultiSelect/index.tsx (hunk 5 above)
Comment (constanzauanini) [reply src/components/common/FilterMultiSelect/index.tsx:56]: Not sure if it is a bug, I think we wanted to work like that. Now that there is a Cancel button, it make more sense to don't clear the parent

File: src/components/common/FilterMultiSelect/index.tsx [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+      )}
+      <Box
+        padding={theme.spacing(3)}
+        width='100%'
+        borderRadius={isMobile ? undefined : theme.spacing(1, 1, 0, 0)}
+        sx={{
+          background: theme.palette.TwClrBg,
+        }}
+      >
+        <MultiSelect
```
Comment (constanzauanini) [inline src/components/common/FilterMultiSelect/index.tsx:97]: Should we add a placeholder here? Maybe "Select..." ?

File: src/components/common/FilterMultiSelect/FilterMultiSelect.test.tsx [hunk 7]
```diff
@@ -0,0 +1,111 @@
+import { fireEvent, render, screen } from '@testing-library/react';
```
Comment (karthikbtf) [inline src/components/common/FilterMultiSelect/FilterMultiSelect.test.tsx]: nice, thanks

File: src/components/common/FilterMultiSelect/FilterMultiSelect.test.tsx [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+  expect(removeButtons.length).toBe(2);
+  fireEvent.click(removeButtons[0]);
+
+  // should see only the option 3 pill
+  expect(screen.queryByText('1')).not.toBeInTheDocument();
+  expect(screen.queryByText('2')).not.toBeInTheDocument();
+  expect(screen.queryByText('3')).toBeInTheDocument();
+
+  // click the apply button
+  const applyButton = screen.getByText('Apply');
```
Comment (karthikbtf) [inline src/components/common/FilterMultiSelect/FilterMultiSelect.test.tsx]: some suggestions for future consideration:
get by id or class - it's more specific in what we are looking for.
same to check for pills, get by class, check count of elements returned.. check html value of elements returned.

**Review by karthikbtf — APPROVED** [review]
🎉

File: src/components/common/FilterMultiSelect/FilterMultiSelect.test.tsx (hunk 8 above)
Comment (charlie-tf) [reply src/components/common/FilterMultiSelect/FilterMultiSelect.test.tsx]: Thanks. I can make those improvements 😊 

Right now having some trouble getting this to run, I committed this by mistake.

**Review by constanzauanini — APPROVED** [review]
👏

**Pull request merged**